pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/AvroSchemaHandler.java

```diff
@@ -55,9 +55,23 @@ public Object deserialize(byte[] bytes) {
     public Object extractField(int index, Object currentRecord) {
         try {
             GenericRecord record = (GenericRecord) currentRecord;
-            return record.get(this.columnHandles.get(index).getPositionIndex());
+            PulsarColumnHandle pulsarColumnHandle = this.columnHandles.get(index);
+            Integer[] positionIndices = pulsarColumnHandle.getPositionIndices();
+            Object curr = record.get(positionIndices[0]);
+            if (curr == null) {
+                return null;
+            }
+            if (positionIndices.length > 0) {
+                for (int i = 1 ; i < positionIndices.length; i++) {
+                    curr = ((GenericRecord) curr).get(positionIndices[i]);
+                    if (curr == null) {
+                        return null;
+                    }
+                }
+            }
+            return curr;
         } catch (Exception ex) {
-            log.error(ex);
+            log.debug(ex,"%s", ex);
```

Comment:
interesting... why we were using airlift logger not slf4j logger?

Reply:
Just to keep things consistent with the presto project, so when we want to move the connector to the presto repo we can do it with little code modification

Reply:
Also I didn't want to introduce more dependencies than necessary and the logger already comes with airlift bootstrap that we need to depend on anyways